server/src/templates/job_detail.html

```diff
@@ -1,81 +1,140 @@
 {% block content %}
-<div class="general-info center">
-    <table>
-        <tr>
-            <th>Job ID:</th>
-            <td>{{ job.job_id }}</td>
-        </tr>
+<div class="p-strip is-shallow">
+    <div class="row">
+        <h1 class="p-heading--3">
+            Job Detail - {{ job.job_id }}
+        </h1>
+    </div>
+</div>
+<table aria-label="Agent table" class="p-table--mobile-card">
+    <tbody>
         <tr>
-            <th>Queue:</th>
+            <td>Queue</td>
             <td>{{ job.job_data.job_queue }}</td>
         </tr>
         <tr>
-            <th>State:</th>
+            <td>State</td>
             <td>{{ job.result_data.job_state }}</td>
         </tr>
         <tr>
-            <th>Created At:</th>
+            <td>Created At</td>
             <td>{{ job.created_at.strftime('%Y-%m-%d %H:%M:%S') }}</td>
         </tr>
-    </table>
-    <br>
-</div>
+    </tbody>
+</table>
 
-<div class="center">
-    <hr>
-    provision_data:
-    <div>
-        {{ job.job_data.provision_data }}
-    </div>
 
-    {% if job.job_data.firmware_update_data %}
-    <hr>
-    firmware_update_data:
-    <div class="center scrollable">
-        <span style="white-space: pre-line">
-            {{ job.job_data.firmware_update_data }}
-        </span>
+<h2 class="p-muted-heading">Job Definition</h2>
+<div class="p-code-snippet">
+    <div class="p-code-snippet__header">
+        <h5 class="p-code-snippet__title">Provision Data</h5>
     </div>
-    {% endif %}
-
-    {% if job.job_data.test_data %}
-    <hr>
-    test_cmds:
-    <div class="center scrollable">
-        <span style="white-space: pre-line">
-            {{ job.job_data.test_data.test_cmds }}
-        </span>
+    <pre class="p-code-snippet__block--numbered language-yaml">
+        <code>
+{{ job.job_data.provision_data }}
+        </code>
+    </pre>
+</div>
+{% if job.job_data.firmware_update_data %}
+<div class="p-code-snippet">
+    <div class="p-code-snippet__header">
+        <h5 class="p-code-snippet__title">Firmware Update Data</h5>
     </div>
-    {% endif %}
+    <pre class="p-code-snippet__block--numbered language-yaml">
+        <code>
+{{ job.job_data.firmware_update_data }}
+        </code>
+    </pre>
+</div>
+{% endif %}
+{% if job.job_data.test_data %}
+<div class="p-code-snippet">
+    <div class="p-code-snippet__header">
+        <h5 class="p-code-snippet__title">Test Commands</h5>
+    </div>
+    <pre class="p-code-snippet__block--numbered">
+        <code>
+{{ job.job_data.test_data.test_cmds }}
+        </code>
+    </pre>
+</div>
+{% endif %}
 
-    <hr>
-    provision_status: {{ job.result_data.provision_status }}<br>
-    provision_output:
-    <div class="center scrollable">
-        <span style="white-space: pre-line">
-            {{ job.result_data.provision_output }}
-        </span>
+<h2 class="p-muted-heading">Provision Phase</h2>
+<span class="p-chip">
+    {% if job.result_data.provision_status %}
+    <span class="p-chip__lead">
```

Comment:
@plars know I review my own work I think I looked at converting these statuses to "chips" - but not sure it can really be called finished. The statuses are I think text descriptions. Maybe this part should be backed out.

Reply:
no, the statuses are basically like an exit code for that stage. So 0 for pass or anything else for fail. However, especially with things like the test phase, it's up to the user on the exit status.

So yeah, thanks for pointing this out. I didn't notice what this change was doing at first. I think this should be removed for now.

Reply:
@jocave I've updated this branch to remove the p-chip spans and replace them with another heading line (to keep it consistent looking) and the numerical exit code if it has it, otherwise it will put `[unknown]`. This is what it looks like now:
![image](https://github.com/canonical/testflinger/assets/1255513/0d6768af-b22e-4bd8-84ca-af95d2148009)

What do you think?

Reply:
The text is great thanks for that. I think I wouldn't personally have it be a header, although I can see that the header that describes the output box does somewhat draw attention.

Reply:
That's nice! The colour helps emphasise the outcome. I think the widgets I selected from the vanilla framework were being a bit stretched to be used as long console output holders anyway